### PR TITLE
Change 'SD card' to 'storage' & cosmetic corrections

### LINKED ISSUE #3920: Wording: 'SD Card' vs. 'Storage'
The wording of several strings is inconsistent regarding the use of 'SD Card' and 'Storage', although all of these strings refer to the very same thing (as seen in the source code).

I'd like to do a small PR correcting that, but want to gather information first what term is prefered.

**Some things to consider:**
- As far as I know, TextSecure is not able to access a 'real' SD card when using KitKat or Lollipop.
- The term 'SD card' could be confusing to users with no SD card slot on their phone. The concept of a 'emulated SD card' could be even more confusing to those users ('There are no power users').

**Possible terms:**
- SD card
- storage
- device storage
- external storage (thx, @2-4601)
- anything else?

I'd appreciate your input!

### PR DESCRIPTION
**Pull request by RiseT**
Fixes #3920 

Change **'SD card'** in strings to **'storage'** for clarity & consistency.

Furthermore, I've done a couple of minor cosmetic string corrections.
I've decided to include these here rather than in a separate PR as there is some overlap with the 'SD card' stuff, and I'd like to save the translators from translating these strings twice.

- Change app's name from 'SMSBackup And Restore' to its official name **'SMS Backup & Restore'**
- Change 'Import encrypted backup' into '**Restore** encrypted backup' as this is the only option that replaces the complete database rather than add to it (which all the 'import' options do). 'restore' is the wording used in the description string as well.
- Remove **redundant periods** at the end of single sentences of some related description strings (in line with Google's design guidelines and the rest of Signal's description strings)
- Remove some **redundant line feeds** in XML (confusing to translators at Transifex)

### CODE REVIEW

**Comment by moxie0** [comment]
you don't have to change the string keys, we'll get those in the course of changing their values over time

**Comment by RiseT** [comment]
@moxie0 Is this OK?

File: res/values/strings.xml [hunk 1]
```diff
@@ -137,7 +137,7 @@
     <string name="ConversationFragment_confirm_message_delete">Confirm message delete</string>
     <string name="ConversationFragment_are_you_sure_you_want_to_permanently_delete_all_selected_messages">Are you sure that you want to permanently delete all selected messages?</string>
     <string name="ConversationFragment_save_to_sd_card">Save to storage?</string>
-    <string name="ConversationFragment_this_media_has_been_stored_in_an_encrypted_database_warning">Saving this media to storage will allow any other apps on your phone to access it.\n\nContinue?</string>
```
Comment (mcginty) [inline res/values/strings.xml]: keep it short
"Saving this media to storage will allow any other apps on your device to access it.\n\nContinue?" gets the job done

File: res/values/strings.xml [hunk 2]
```diff
@@ -196,15 +196,13 @@
     <string name="ExperienceUpgradeActivity_textsecure_is_now_signal_long">TextSecure and RedPhone are now one app: Signal. Tap to explore.</string>
     <!-- ExportFragment -->
     <string name="ExportFragment_export">Export</string>
-    <string name="ExportFragment_export_plaintext_to_sd_card">Export plaintext to SD card?</string>
-    <string name="ExportFragment_warning_this_will_export_the_plaintext_contents">Warning, this will
-        export the plaintext contents of your Signal messages to the SD card.
-    </string>
+    <string name="ExportFragment_export_plaintext_to_storage">Export plaintext to storage?</string>
+    <string name="ExportFragment_warning_this_will_export_the_plaintext_contents">Warning, this will export the plaintext contents of your Signal messages to the device's storage.</string>
```
Comment (RiseT) [inline res/values/strings.xml]: @mcginty  Should I remove 'the device's" here as well?

File: res/values/strings.xml [hunk 3]
```diff
@@ -624,19 +622,13 @@
     <string name="database_upgrade_activity__updating_database">Updating database...</string>
 
     <string name="export_fragment__export_plaintext_backup">Export plaintext backup</string>
-    <string name="export_fragment__export_a_plaintext_backup_compatible_with">
-        Export a plaintext backup compatible with \'SMSBackup And Restore\' to the SD card.</string>
+    <string name="export_fragment__export_a_plaintext_backup_compatible_with">Export a plaintext backup compatible with \'SMS Backup &amp; Restore\' to the device's storage</string>
```
Comment (RiseT) [inline res/values/strings.xml]: @mcginty And here?

File: res/values/strings.xml (hunk 1 above)
Comment (RiseT) [inline res/values/strings.xml]: OK.

**Comment by RiseT** [comment]
@mcginty As I suspect that the answer to my question will be 'yes', I've removed all 3 occurences of 'the device's' with the latest commit and thus shortened all of those strings.

**Comment by moxie0** [comment]
Where did "storage" come from? I feel like I'm used to seeing "external storage," is that not a thing anymore?

**Comment by RiseT** [comment]
@moxie0 See my initial question & subsequent discussion here: #3920

**Comment by moxie0** [comment]
ok cool but you gotta squash these commits and rebase onto 3.3.0

**Pull request closed without merging**